tests/osgi-tests/src/test/java/org/wso2/carbon/osgi/config/PropertyResolveConfigProviderOSGITest.java

```diff
@@ -0,0 +1,120 @@
+@ExamReactorStrategy(PerClass.class)
+@ExamFactory(CarbonContainerFactory.class)
+public class PropertyResolveConfigProviderOSGITest {
+
+    private static final String DEPLOYMENT_FILENAME = "deployment.yaml";
+    private static final String SECRET_PROPERTIES_FILENAME = "secrets.properties";
+
+    @Inject
+    ConfigProvider configProvider;
+
+    @Configuration
+    public Option[] createConfiguration() {
+        return new Option[]{copyCarbonYAMLOption(), systemProperty("server.id").value("test-carbon-kernel"),
+                setEnvPropertyOption("server.name", "Test WSO2 Carbon Kernel"), copySecretPropertiesOption()
+        };
+    }
+
+    @Test
+    public void testPropertyResolve() throws ConfigurationException {
+        CarbonConfiguration carbonConfiguration = configProvider.getConfigurationObject(
+                CarbonConfiguration.class);
+        String id = carbonConfiguration.getId();
+        String name = carbonConfiguration.getName();
+        String tenant = carbonConfiguration.getTenant();
+
+        Assert.assertEquals(id, "test-carbon-kernel", "id should be test-carbon-kernel");
+        Assert.assertEquals(name, "Test WSO2 Carbon Kernel",
+                "name should be Test WSO2 Carbon Kernel");
+        Assert.assertEquals(tenant, "default", "tenant should be default");
+
+    }
+
+    /**
+     * Replace the existing deployment.yaml file with the file found at property-resolve directory.
+     */
+    private CopyFileOption copyCarbonYAMLOption() {
+
+        Path carbonYmlFilePath;
+
+        String basedir = System.getProperty("basedir");
+        if (basedir == null) {
+            basedir = Paths.get(".").toString();
+        }
+        carbonYmlFilePath = Paths.get(basedir, "src", "test", "resources", "property-resolve",
+                DEPLOYMENT_FILENAME);
+        return new CopyFileOption(carbonYmlFilePath, Paths.get("conf", DEPLOYMENT_FILENAME));
+    }
+
+    /**
+     * Replace the existing secrets.properties file with the file found in property-resolve directory.
+     *
+     * @return CopyFileOption object.
+     */
+    private CopyFileOption copySecretPropertiesOption() {
+        Path carbonYmlFilePath;
+
+        String basedir = System.getProperty("basedir");
+        if (basedir == null) {
+            basedir = Paths.get(".").toString();
+        }
+        carbonYmlFilePath = Paths.get(basedir, "src", "test", "resources", "property-resolve",
+                SECRET_PROPERTIES_FILENAME);
+        return new CopyFileOption(carbonYmlFilePath, Paths.get("conf", SECRET_PROPERTIES_FILENAME));
+    }
+
+    /**
+     *  Set environment property.
+     *
+     * @param key key to set in environment property.
+     * @param value value to set for the above mentioned key.
+     * @return empty Option object.
+     */
+    private Option setEnvPropertyOption(String key, String value) {
+        EnvironmentUtils.setEnv(key, value);
+        return new Option() {
+        };
```

Comment:
It is because I wanted to add environment property inside createConfiguration method. So I add it in return array. Anyway since this is not an option we can add it method itself. so fixed in 7a068c494b4822c8afbd915e9d9094b4da4d5c1d